src/trip.core.js

```diff
@@ -507,7 +507,7 @@ Trip.prototype = {
     this.setTripBlock(o);
     this.showTripBlock(o);
 
-    if (this.settings.enableAnimation) {
+    if (TripUtils.isSet(o.enableAnimation, this.settings.enableAnimation)) {
```

Comment:
so we always remove previous animation classes and decide whether we need to add it back or not based on current state.